lib/ruby_lsp/requests/path_completion.rb

```diff
@@ -22,33 +22,24 @@ class PathCompletion < Listener
       sig { override.returns(ResponseType) }
       attr_reader :response
 
-      sig { params(emitter: EventEmitter, message_queue: Thread::Queue).void }
-      def initialize(emitter, message_queue)
-        super
+      sig { params(index: RubyIndexer::Index, emitter: EventEmitter, message_queue: Thread::Queue).void }
+      def initialize(index, emitter, message_queue)
+        super(emitter, message_queue)
         @response = T.let([], ResponseType)
-        @tree = T.let(RubyIndexer::PrefixTree.new(collect_load_path_files), RubyIndexer::PrefixTree)
+        @index = index
 
         emitter.register(self, :on_tstring_content)
       end
 
       sig { params(node: SyntaxTree::TStringContent).void }
       def on_tstring_content(node)
-        @tree.search(node.value).sort.each do |path|
+        @index.search_require_paths(node.value).sort!.each do |path|
```

Comment:
Are we starting to recommend in place operations to avoid unnecessary array creations?

Also will this be a hot path? Is there a way we can sort only if the contents have changed?

Reply:
Awesome, thanks for the context